### PR TITLE
Upgrade Model-Binding to 6.0

### LINKED ISSUE #23763: .NET 6 Model Binding
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e24f48f8-6c4d-ecda-6ce6-a7826d396e99
* Version Independent ID: 9c4af557-8d06-4c6c-ad2a-0dd234547563
* Content: [Model Binding in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-5.0)
* Content Source: [aspnetcore/mvc/models/model-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/models/model-binding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #23763.

[Internal Preview URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-6.0&branch=pr-en-us-24019).

### CODE REVIEW

File: aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/Program.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+namespace ModelBindingSample.Snippets;
+
+public static class Program
+{
+    public static void AddValueProviderFactory(WebApplicationBuilder builder)
+    {
+        // <snippet_AddValueProviderFactory>
+        builder.Services.AddControllers(options =>
+        {
+            options.ValueProviderFactories.Add(new CookieValueProviderFactory());
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/Program.cs:13]: Much better than the previous sample that included a bunch of unrelated code.

File: aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/Program.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        // </snippet_AddValueProviderFactory>
+    }
+
+    public static void ReplaceQueryStringValueProviderFactory(WebApplicationBuilder builder)
+    {
+        // <snippet_ReplaceQueryStringValueProviderFactory>
+        builder.Services.AddControllers(options =>
+        {
+            var index = options.ValueProviderFactories.IndexOf(
+                options.ValueProviderFactories.OfType<QueryStringValueProviderFactory>().Single());
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/Program.cs:24]:
```suggestion
             options.ValueProviderFactories.OfType<QueryStringValueProviderFactory>().Single());
```
This suggestion will remove the horizontal scroll bar.

File: aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/ObjectIdConverter.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace ModelBindingSample.Snippets;
+
+// <snippet_Class>
+internal class ObjectIdConverter : JsonConverter<ObjectId>
+{
+    public override ObjectId Read(
+            ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/ObjectIdConverter.cs:10]:
```suggestion
        ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
```

**Review by Rick-Anderson — APPROVED** [review]
Much better. Much cleaner snippets.

File: aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/ObjectIdConverter.cs (hunk 3 above)
Comment (Rick-Anderson) [reply aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/ObjectIdConverter.cs:10]: And those are just suggestions. If you think the code looks better with the horizontal scroll bar, that's fine.

**Comment by serpent5** [comment]
@Rick-Anderson Thanks for the suggestions. I often forget to check the width of the code snippets in the preview, but I'm working on that...
